neurom/core/_neuron.py

```diff
@@ -39,10 +39,17 @@
 from neurom.core.dataformat import COLS
 from neurom.utils import memoize
 
-from . import NeuriteType, Tree
+from . import NeuriteType, Tree, NeuriteIter
 
+# NRN simulator iteration order
+NRN_ORDER = {NeuriteType.soma: 0,
+             NeuriteType.axon: 1,
+             NeuriteType.basal_dendrite: 2,
+             NeuriteType.apical_dendrite: 3,
+             NeuriteType.undefined: 4}
 
-def iter_neurites(obj, mapfun=None, filt=None):
+
+def iter_neurites(obj, mapfun=None, filt=None, iterator_type=NeuriteIter.FileOrder):
```

Comment:
ok, I see why I got confused: the `iterator_type` is being used in two different ways: the old way - to dispatch one of Tree.i*order, but now it also is used to trigger what kind of ordering is used for NRN order.  I think this is confusing.

Since `tree.py` is being modified (which I think is leaking an abstraction, personally), having ifileorder there would mean that the non-symmetric dispatch isn't needed.

I wonder whether if having an 'iterate like NEURON' should really just be a separate bare function that you can call, so if people need it, they can directly call it, but this portion of the code, and the rest of the iteration mechanisms don't have this.